### ISSUE TITLE
quit-window is not advised

### ISSUE THREAD

**Issue by hrehfeld** [raxod502/radian#492]
This happens frequently:

- sometimes when opening files via consult
- when composing messages in mu4e

(I will write down more cases over time)

Because of #493 , I have disabled gitconfig-mode and gitignore-mode in radian.el, could be related?

```
Debugger entered--Lisp error: (error "ad-deactivate: ‘quit-window’ is not advised")
  error("ad-deactivate: `%s' is not advised" quit-window)
  ad-deactivate(quit-window)
  (progn (setq git-gutter:ask-p nil) (global-git-gutter-mode 1) (defalias 'radian-git-gutter:beginning-of-hunk #'(lambda nil "Move to beginning of current diff hunk." (interactive) (git-gutter:awhen (git-gutter:search-here-diffinfo git-gutter:diffinfos) (let ((lines ...)) (forward-line lines))))) (remove-hook 'post-command-hook #'git-gutter:post-command-hook) (ad-deactivate #'quit-window) (ad-deactivate #'switch-to-buffer) (defvar radian--git-gutter-last-buffer-and-window nil "Cons of current buffer and selected window before ...") (progn (defalias 'radian--git-gutter-on-buffer-or-window-change #'(lambda nil "Update `git-gutter' when current buffer or selecte..." (let ((new ...)) (if (equal new radian--git-gutter-last-buffer-and-window) nil (setq radian--git-gutter-last-buffer-and-window new) (save-current-buffer ... ...))))) (let ((--dolist-tail-- '(post-command-hook))) (while --dolist-tail-- (let ((hook (car --dolist-tail--))) (add-hook hook 'radian--git-gutter-on-buffer-or-window-change) (setq --dolist-tail-- (cdr --dolist-tail--)))))) (progn (defvar use-package--warning120 #'(lambda (keyword err) (let ((msg ...)) (display-warning 'use-package msg :error)))) (condition-case err (eval-after-load 'autorevert #'(lambda nil (condition-case err (progn ... t) (... ...)))) ((debug error) (funcall use-package--warning120 :catch err)))) (progn (defvar use-package--warning121 #'(lambda (keyword err) (let ((msg ...)) (display-warning 'use-package msg :error)))) (condition-case err (eval-after-load 'apheleia #'(lambda nil (condition-case err (progn ... t) (... ...)))) ((debug error) (funcall use-package--warning121 :catch err)))) (blackout 'git-gutter-mode nil) t)
  (condition-case err (progn (setq git-gutter:ask-p nil) (global-git-gutter-mode 1) (defalias 'radian-git-gutter:beginning-of-hunk #'(lambda nil "Move to beginning of current diff hunk." (interactive) (git-gutter:awhen (git-gutter:search-here-diffinfo git-gutter:diffinfos) (let (...) (forward-line lines))))) (remove-hook 'post-command-hook #'git-gutter:post-command-hook) (ad-deactivate #'quit-window) (ad-deactivate #'switch-to-buffer) (defvar radian--git-gutter-last-buffer-and-window nil "Cons of current buffer and selected window before ...") (progn (defalias 'radian--git-gutter-on-buffer-or-window-change #'(lambda nil "Update `git-gutter' when current buffer or selecte..." (let (...) (if ... nil ... ...)))) (let ((--dolist-tail-- '(post-command-hook))) (while --dolist-tail-- (let ((hook ...)) (add-hook hook 'radian--git-gutter-on-buffer-or-window-change) (setq --dolist-tail-- (cdr --dolist-tail--)))))) (progn (defvar use-package--warning120 #'(lambda (keyword err) (let (...) (display-warning ... msg :error)))) (condition-case err (eval-after-load 'autorevert #'(lambda nil (condition-case err ... ...))) ((debug error) (funcall use-package--warning120 :catch err)))) (progn (defvar use-package--warning121 #'(lambda (keyword err) (let (...) (display-warning ... msg :error)))) (condition-case err (eval-after-load 'apheleia #'(lambda nil (condition-case err ... ...))) ((debug error) (funcall use-package--warning121 :catch err)))) (blackout 'git-gutter-mode nil) t) ((debug error) (funcall use-package--warning119 :config err)))
  (closure (bootstrap-version straight-current-profile t) nil (condition-case err (progn (setq git-gutter:ask-p nil) (global-git-gutter-mode 1) (defalias 'radian-git-gutter:beginning-of-hunk #'(lambda nil "Move to beginning of current diff hunk." (interactive) (git-gutter:awhen (git-gutter:search-here-diffinfo git-gutter:diffinfos) (let ... ...)))) (remove-hook 'post-command-hook #'git-gutter:post-command-hook) (ad-deactivate #'quit-window) (ad-deactivate #'switch-to-buffer) (defvar radian--git-gutter-last-buffer-and-window nil "Cons of current buffer and selected window before ...") (progn (defalias 'radian--git-gutter-on-buffer-or-window-change #'(lambda nil "Update `git-gutter' when current buffer or selecte..." (let ... ...))) (let ((--dolist-tail-- '...)) (while --dolist-tail-- (let (...) (add-hook hook ...) (setq --dolist-tail-- ...))))) (progn (defvar use-package--warning120 #'(lambda (keyword err) (let ... ...))) (condition-case err (eval-after-load 'autorevert #'(lambda nil ...)) ((debug error) (funcall use-package--warning120 :catch err)))) (progn (defvar use-package--warning121 #'(lambda (keyword err) (let ... ...))) (condition-case err (eval-after-load 'apheleia #'(lambda nil ...)) ((debug error) (funcall use-package--warning121 :catch err)))) (blackout 'git-gutter-mode nil) t) ((debug error) (funcall use-package--warning119 :config err))))()
  eval-after-load-helper("/home/hrehfeld/.emacs.d/straight/build/git-gutter/...")
  do-after-load-evaluation("/home/hrehfeld/.emacs.d/straight/build/git-gutter/...")
  require(git-gutter)
  radian--git-gutter-load()
  run-hooks(find-file-hook)
  after-find-file(nil t)
  find-file-noselect-1(#<buffer bookmark-default.el/var/.emacs.d/hrehfeld/home/> "~/.emacs.d/var/bookmark-default.el" nil nil "~/.emacs.d/var/bookmark-default.el" (61664412 65025))
  find-file-noselect("/home/hrehfeld/.emacs.d/var/bookmark-default.el")
  #<subr bookmark-load>("/home/hrehfeld/.emacs.d/var/bookmark-default.el" t t nil)
  apply(#<subr bookmark-load> ("/home/hrehfeld/.emacs.d/var/bookmark-default.el" t t))
  (progn (fset #'message #'ignore) (apply func args))
  (unwind-protect (progn (fset #'message #'ignore) (apply func args)) (fset #'message old))
  (let* ((old (symbol-function #'message))) (unwind-protect (progn (fset #'message #'ignore) (apply func args)) (fset #'message old)))
  radian--advice-silence-messages(#<subr bookmark-load> "/home/hrehfeld/.emacs.d/var/bookmark-default.el" t t)
  apply(radian--advice-silence-messages #<subr bookmark-load> ("/home/hrehfeld/.emacs.d/var/bookmark-default.el" t t))
  bookmark-load("/home/hrehfeld/.emacs.d/var/bookmark-default.el" t t)
  bookmark-maybe-load-default-file()
  bookmark-all-names()
  #f(compiled-function (src) #<bytecode -0x2c22fbc4da1b0dc>)((:name "Bookmark" :narrow 109 :category bookmark :face consult-bookmark :history bookmark-history :items bookmark-all-names :state consult--bookmark-state))
  mapc(#f(compiled-function (src) #<bytecode -0x2c22fbc4da1b0dc>) [(:name "Hidden Buffer" :narrow 32 :hidden t :category buffer :face consult-buffer :history buffer-name-history :action consult--buffer-action :items #f(compiled-function () #<bytecode -0x1542f36f3541b2cf>)) (:name "Buffer" :narrow 98 :category buffer :face consult-buffer :history buffer-name-history :state consult--buffer-state :default t :items #f(compiled-function () #<bytecode 0x14d921fe794c4>)) (:name "File" :narrow 102 :category file :face consult-file :history file-name-history :state consult--file-state :enabled #f(compiled-function () #<bytecode 0x8f54d19f39d5>) :items #f(compiled-function () #<bytecode 0x1a6207979c59aeff>)) (:name "Bookmark" :narrow 109 :category bookmark :face consult-bookmark :history bookmark-history :items bookmark-all-names :state consult--bookmark-state) (:name "Project Buffer" :narrow (112 . "Project") :hidden t :category buffer :face consult-buffer :history buffer-name-history :state consult--buffer-state :enabled #f(compiled-function () #<bytecode 0x1f401c010471>) :items #f(compiled-function () #<bytecode -0x19af76fa68aacccf>)) (:name "Project File" :narrow (112 . "Project") :hidden t :category file :face consult-file :history file-name-history :state consult--file-state :enabled #f(compiled-function () #<bytecode 0x8f49847bb3c5>) :items #f(compiled-function () #<bytecode -0x17b0aa5d84218af6>))])
  seq-do(#f(compiled-function (src) #<bytecode -0x2c22fbc4da1b0dc>) [(:name "Hidden Buffer" :narrow 32 :hidden t :category buffer :face consult-buffer :history buffer-name-history :action consult--buffer-action :items #f(compiled-function () #<bytecode -0x1542f36f3541b2cf>)) (:name "Buffer" :narrow 98 :category buffer :face consult-buffer :history buffer-name-history :state consult--buffer-state :default t :items #f(compiled-function () #<bytecode 0x14d921fe794c4>)) (:name "File" :narrow 102 :category file :face consult-file :history file-name-history :state consult--file-state :enabled #f(compiled-function () #<bytecode 0x8f54d19f39d5>) :items #f(compiled-function () #<bytecode 0x1a6207979c59aeff>)) (:name "Bookmark" :narrow 109 :category bookmark :face consult-bookmark :history bookmark-history :items bookmark-all-names :state consult--bookmark-state) (:name "Project Buffer" :narrow (112 . "Project") :hidden t :category buffer :face consult-buffer :history buffer-name-history :state consult--buffer-state :enabled #f(compiled-function () #<bytecode 0x1f401c010471>) :items #f(compiled-function () #<bytecode -0x19af76fa68aacccf>)) (:name "Project File" :narrow (112 . "Project") :hidden t :category file :face consult-file :history file-name-history :state consult--file-state :enabled #f(compiled-function () #<bytecode 0x8f49847bb3c5>) :items #f(compiled-function () #<bytecode -0x17b0aa5d84218af6>))])
  consult--multi-candidates([(:name "Hidden Buffer" :narrow 32 :hidden t :category buffer :face consult-buffer :history buffer-name-history :action consult--buffer-action :items #f(compiled-function () #<bytecode -0x1542f36f3541b2cf>)) (:name "Buffer" :narrow 98 :category buffer :face consult-buffer :history buffer-name-history :state consult--buffer-state :default t :items #f(compiled-function () #<bytecode 0x14d921fe794c4>)) (:name "File" :narrow 102 :category file :face consult-file :history file-name-history :state consult--file-state :enabled #f(compiled-function () #<bytecode 0x8f54d19f39d5>) :items #f(compiled-function () #<bytecode 0x1a6207979c59aeff>)) (:name "Bookmark" :narrow 109 :category bookmark :face consult-bookmark :history bookmark-history :items bookmark-all-names :state consult--bookmark-state) (:name "Project Buffer" :narrow (112 . "Project") :hidden t :category buffer :face consult-buffer :history buffer-name-history :state consult--buffer-state :enabled #f(compiled-function () #<bytecode 0x1f401c010471>) :items #f(compiled-function () #<bytecode -0x19af76fa68aacccf>)) (:name "Project File" :narrow (112 . "Project") :hidden t :category file :face consult-file :history file-name-history :state consult--file-state :enabled #f(compiled-function () #<bytecode 0x8f49847bb3c5>) :items #f(compiled-function () #<bytecode -0x17b0aa5d84218af6>))])
  consult--multi((consult--source-hidden-buffer consult--source-buffer consult--source-recent-file consult--source-bookmark consult--source-project-buffer consult--source-project-recent-file) :require-match confirm-after-completion :prompt "Switch to: " :history consult--buffer-history :sort nil)
  consult-buffer()
  funcall-interactively(consult-buffer)
  command-execute(consult-buffer)

```

```
Debugger entered--Lisp error: (error "ad-deactivate: ‘quit-window’ is not advised")
  error("ad-deactivate: `%s' is not advised" quit-window)
  ad-deactivate(quit-window)
  (progn (setq git-gutter:ask-p nil) (global-git-gutter-mode 1) (defalias 'radian-git-gutter:beginning-of-hunk #'(lambda nil "Move to beginning of current diff hunk." (interactive) (git-gutter:awhen (git-gutter:search-here-diffinfo git-gutter:diffinfos) (let ((lines ...)) (forward-line lines))))) (remove-hook 'post-command-hook #'git-gutter:post-command-hook) (ad-deactivate #'quit-window) (ad-deactivate #'switch-to-buffer) (defvar radian--git-gutter-last-buffer-and-window nil "Cons of current buffer and selected window before ...") (progn (defalias 'radian--git-gutter-on-buffer-or-window-change #'(lambda nil "Update `git-gutter' when current buffer or selecte..." (let ((new ...)) (if (equal new radian--git-gutter-last-buffer-and-window) nil (setq radian--git-gutter-last-buffer-and-window new) (save-current-buffer ... ...))))) (let ((--dolist-tail-- '(post-command-hook))) (while --dolist-tail-- (let ((hook (car --dolist-tail--))) (add-hook hook 'radian--git-gutter-on-buffer-or-window-change) (setq --dolist-tail-- (cdr --dolist-tail--)))))) (progn (defvar use-package--warning120 #'(lambda (keyword err) (let ((msg ...)) (display-warning 'use-package msg :error)))) (condition-case err (eval-after-load 'autorevert #'(lambda nil (condition-case err (progn ... t) (... ...)))) ((debug error) (funcall use-package--warning120 :catch err)))) (progn (defvar use-package--warning121 #'(lambda (keyword err) (let ((msg ...)) (display-warning 'use-package msg :error)))) (condition-case err (eval-after-load 'apheleia #'(lambda nil (condition-case err (progn ... t) (... ...)))) ((debug error) (funcall use-package--warning121 :catch err)))) (blackout 'git-gutter-mode nil) t)
  (condition-case err (progn (setq git-gutter:ask-p nil) (global-git-gutter-mode 1) (defalias 'radian-git-gutter:beginning-of-hunk #'(lambda nil "Move to beginning of current diff hunk." (interactive) (git-gutter:awhen (git-gutter:search-here-diffinfo git-gutter:diffinfos) (let (...) (forward-line lines))))) (remove-hook 'post-command-hook #'git-gutter:post-command-hook) (ad-deactivate #'quit-window) (ad-deactivate #'switch-to-buffer) (defvar radian--git-gutter-last-buffer-and-window nil "Cons of current buffer and selected window before ...") (progn (defalias 'radian--git-gutter-on-buffer-or-window-change #'(lambda nil "Update `git-gutter' when current buffer or selecte..." (let (...) (if ... nil ... ...)))) (let ((--dolist-tail-- '(post-command-hook))) (while --dolist-tail-- (let ((hook ...)) (add-hook hook 'radian--git-gutter-on-buffer-or-window-change) (setq --dolist-tail-- (cdr --dolist-tail--)))))) (progn (defvar use-package--warning120 #'(lambda (keyword err) (let (...) (display-warning ... msg :error)))) (condition-case err (eval-after-load 'autorevert #'(lambda nil (condition-case err ... ...))) ((debug error) (funcall use-package--warning120 :catch err)))) (progn (defvar use-package--warning121 #'(lambda (keyword err) (let (...) (display-warning ... msg :error)))) (condition-case err (eval-after-load 'apheleia #'(lambda nil (condition-case err ... ...))) ((debug error) (funcall use-package--warning121 :catch err)))) (blackout 'git-gutter-mode nil) t) ((debug error) (funcall use-package--warning119 :config err)))
  (closure (bootstrap-version straight-current-profile t) nil (condition-case err (progn (setq git-gutter:ask-p nil) (global-git-gutter-mode 1) (defalias 'radian-git-gutter:beginning-of-hunk #'(lambda nil "Move to beginning of current diff hunk." (interactive) (git-gutter:awhen (git-gutter:search-here-diffinfo git-gutter:diffinfos) (let ... ...)))) (remove-hook 'post-command-hook #'git-gutter:post-command-hook) (ad-deactivate #'quit-window) (ad-deactivate #'switch-to-buffer) (defvar radian--git-gutter-last-buffer-and-window nil "Cons of current buffer and selected window before ...") (progn (defalias 'radian--git-gutter-on-buffer-or-window-change #'(lambda nil "Update `git-gutter' when current buffer or selecte..." (let ... ...))) (let ((--dolist-tail-- '...)) (while --dolist-tail-- (let (...) (add-hook hook ...) (setq --dolist-tail-- ...))))) (progn (defvar use-package--warning120 #'(lambda (keyword err) (let ... ...))) (condition-case err (eval-after-load 'autorevert #'(lambda nil ...)) ((debug error) (funcall use-package--warning120 :catch err)))) (progn (defvar use-package--warning121 #'(lambda (keyword err) (let ... ...))) (condition-case err (eval-after-load 'apheleia #'(lambda nil ...)) ((debug error) (funcall use-package--warning121 :catch err)))) (blackout 'git-gutter-mode nil) t) ((debug error) (funcall use-package--warning119 :config err))))()
  eval-after-load-helper("/home/hrehfeld/.emacs.d/straight/build/git-gutter/...")
  do-after-load-evaluation("/home/hrehfeld/.emacs.d/straight/build/git-gutter/...")
  require(git-gutter)
  radian--git-gutter-load()
  run-hooks(find-file-hook)
  after-find-file(t t)
 <retracted>
  mu4e-draft-open(new nil nil)
  mu4e~compose-handler(new)
  mu4e-compose(new)
  mu4e-compose-new()
  funcall-interactively(mu4e-compose-new)
  command-execute(mu4e-compose-new)
```

**Comment by hrehfeld** [comment]
Was this fixed in 0e480c522?

**Comment by raxod502** [comment]
Yes, I think so. But let me know if the problem reoccurs.
